### PR TITLE
Create logfile directory & set sane permissions

### PR DESCRIPTION
**Pull request by jgeusebroek**

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -38,10 +38,14 @@
   notify: Restart Caddy
 
 - name: Create directories
-  file: path={{ item }} state=directory owner={{ caddy_user }} mode=0700
+  file: path={{ item.dir }} state=directory owner={{ item.user|default(caddy_user) }} group={{ item.group|default('root') }} mode={{ item.mode|default(750) }}
```
Comment (lbischof) [inline tasks/main.yml]: Why not group caddy_user? Why complicate everything with defaults? Can't you just add `"{{ caddy_log_dir }}"` to `with_items`?

**Comment by lbischof** [comment]
Not too sure if this is a good idea. Docker should log to stdout

**Comment by jgeusebroek** [comment]
The default will stay stdout, no change there. It only provides the possibility to log to /var/log/caddy/*. This could be for the caddy daemon itself, but also access/error logs.

File: tasks/main.yml (hunk 1 above)
Comment (jgeusebroek) [inline tasks/main.yml]: True, it is more complicated. But most of the time though I see log directories in /var/log owned by the daemon user, and group root.

But as root has access already, it shouldn't matter much. Could be reverted if you wish :)

**Comment by lbischof** [comment]
Are you using the image and need this? If not, then I would prefer to log everything to stdout

**Comment by jgeusebroek** [comment]
I am using the role, not containerized.  I already create the log directory separate from the role in my own playbook, so not strictly necessary for me.

If it was my role, I'd add it, if only just for the ability to choose. The default will be the same as before. The only downside is the /var/log/caddy directory will be there even when you're not using it.

The directory permission and group change, is of course optional. I can change that if you prefer.
Should you decide not to add this option, no problem!

**Comment by lbischof** [comment]
Sorry, I was doing something else with Docker and confused the two. I will merge this, but please simplify the task and remove the default methods.

**Comment by jgeusebroek** [comment]
Done ;)

**Comment by lbischof** [comment]
Thanks :+1:

**Comment by jgeusebroek** [comment]
Seems like you merged, but it didn't include the last commit? Not sure what happened.

**Comment by lbischof** [comment]
Not sure what you mean? Are changes missing? https://github.com/antoiner77/caddy-ansible/commit/8ea397955368178b164b40f62c22283dd82623e1

**Comment by jgeusebroek** [comment]
My bad, I misread :)

**Pull request merged**